api/cloudprotocol/desiredstatus.go

```diff
@@ -0,0 +1,225 @@
+
+// DesiredStatusMessageType desired status message type.
+const DesiresStatusMessageType = "desiredStatus"
+
+// SOTA/FOTA schedule type.
+const (
+	ForceUpdate     = "force"
+	TriggerUpdate   = "trigger"
+	TimetableUpdate = "timetable"
+)
+
+/***********************************************************************************************************************
+ * Types
+ **********************************************************************************************************************/
+
+// FileSystemMount specifies a mount instructions.
+type FileSystemMount struct {
+	Destination string   `json:"destination"`
+	Type        string   `json:"type,omitempty"`
+	Source      string   `json:"source,omitempty"`
+	Options     []string `json:"options,omitempty"`
+}
+
+// HostInfo struct represents entry in /etc/hosts.
+type HostInfo struct {
+	IP       string `json:"ip"`
+	Hostname string `json:"hostname"`
+}
+
+// DeviceInfo device information.
+type DeviceInfo struct {
+	Name        string   `json:"name"`
+	SharedCount int      `json:"sharedCount,omitempty"`
+	Groups      []string `json:"groups,omitempty"`
+	HostDevices []string `json:"hostDevices"`
+}
+
+// ResourceInfo resource information.
+type ResourceInfo struct {
+	Name   string            `json:"name"`
+	Groups []string          `json:"groups,omitempty"`
+	Mounts []FileSystemMount `json:"mounts,omitempty"`
+	Env    []string          `json:"env,omitempty"`
+	Hosts  []HostInfo        `json:"hosts,omitempty"`
+}
+
+// ResourceOvercommits resource overcommits.
+type ResourceOvercommits struct {
+	CPU     float32 `json:"cpu"`
+	Mem     float32 `json:"mem"`
+	Storage float32 `json:"storage"`
+}
+
+// NodeConfig node configuration.
+type NodeConfig struct {
+	NodeID              string               `json:"nodeId"`
+	NodeType            string               `json:"nodeType"`
+	ResourceOvercommits ResourceOvercommits  `json:"overcommits"`
+	AlertRules          *aostypes.AlertRules `json:"alertRules,omitempty"`
+	Devices             []DeviceInfo         `json:"devices,omitempty"`
+	Resources           []ResourceInfo       `json:"resources,omitempty"`
+	Labels              []string             `json:"labels,omitempty"`
+	Priority            uint32               `json:"priority,omitempty"`
+}
+
+// UnitConfig unit configuration.
+type UnitConfig struct {
+	FormatVersion string       `json:"formatVersion"`
+	Version       string       `json:"version"`
+	Nodes         []NodeConfig `json:"nodes"`
+}
+
+// DecryptionInfo update decryption info.
+type DecryptionInfo struct {
+	BlockAlg     string `json:"blockAlg"`
+	BlockIv      []byte `json:"blockIv"`
+	BlockKey     []byte `json:"blockKey"`
+	AsymAlg      string `json:"asymAlg"`
+	ReceiverInfo *struct {
+		Serial string `json:"serial"`
+		Issuer []byte `json:"issuer"`
+	} `json:"receiverInfo"`
+}
+
+// DownloadInfo struct contains how to download item.
```

Comment:
it looks like it specifies how to download mulitple items

Reply:
Unfortunately or fortunately, this information for one item.